roles/setup_repo/tasks/PG_RedHat_setuprepos.yml

```diff
@@ -149,12 +151,33 @@
   failed_when: >
     reposub.rc != 0 or 'error: ' in reposub.stdout.lower()
   changed_when: reposub.rc == '0'
-  when: >
-    tpa_subscription_token|length > 0
-    and install_bdr_packages|bool
-    and pg_version|int >= 14
+  when:
+    - enable_edb_repo|bool
+    - tpa_subscription_token|length > 0
+    - install_bdr_packages|bool
+    - pg_version|int == 14
+    - bdr_version|int == 4
   loop: "{{ edb_2q_repositories }}"
 
+- name: Install PGD packages repo
+  ansible.builtin.yum_repository:
+    name: "enterprisedb-postgres_distributed-{{ item }}"
+    description: "enterprisedb-postgres_distributed-{{ item }}"
+    file: "enterprisedb-postgres_distributed"
+    baseurl: "{{ edb_repo_base_download }}/postgres_distributed/rpm/el/{{ ansible_distribution_major_version }}/{{ item }}"
+    gpgkey: "{{ edb_repo_base_download }}/postgres_distributed/gpg.key"
```

Comment:
I think these should be a variable defined in `main.yml`, similar to other repo url's and their corresponding gpg key's. Something like `edb_pgd_rpm_repo_8`, `edb_pgd_rpm_repo_7`,  `edb_pgd_apt_repo` and `edb_pgd_gpg_key`?

Reply:
I believe using the ansible_distribution_major_version variable is a good idea instead of making it specific to the OS version. I can see the advantages of having both the edb_pgd_rpm_repo and edb_repo_base_download_gpg_key.

Reply:
I made several changes and created a new array variable named pgd_rpm_repo. This variable encompasses the repository name, description, and repository link for both x86_64 and noarch. I used the same code using pgd_rpm_repo. By doing this, we guarantee that we avoid duplicating the code and declare it in defaults/main.yml.